internal/providers/http/http.go

```diff
@@ -36,9 +36,6 @@ type REST struct {
 	credential provifv1.RestCredential
 }
 
-// Ensure that REST implements the REST interface
-var _ provifv1.REST = (*REST)(nil)
```

Comment:
why do we need to remove these?

Reply:
so, REST and Git are not actually providers, but helper code that providers use to implement the relevant traits. I was thinking it would be good to start migrating away from these "trying" to implement the provider interface. This gets a little tricky, as we do treat them as providers in some test code, so I might just create a provider wrapper that uses these.